### PR TITLE
Set Architecture of containerdisks

### PR DESCRIPTION
**Pull request by 0xFelix**
**What this PR does / why we need it**:

This sets the Architecture in config files of containerdisk images to 'amd64'.
Previously the Architecture was empty and tools like 'ctr' failed to import containerdisks because of it.

See https://groups.google.com/d/msgid/kubevirt-dev/4534a4f5-178e-46c2-90ee-ef60a216c7ecn%40googlegroups.com

**Release note**:

```release-note
Set Architecture of containerdisks to amd64
```

### CODE REVIEW

**Review by lyarwood — APPROVED** [review]

File: pkg/build/build.go [hunk 1]
```diff
@@ -25,6 +26,17 @@ func BuildContainerDisk(imgPath string, checksum string) (v1.Image, error) {
 		return nil, fmt.Errorf("error appending the image layer: %v", err)
 	}
 
+	cf, err := img.ConfigFile()
+	if err != nil {
+		return nil, fmt.Errorf("error getting the image config file: %v", err)
+	}
+
+	cf.Architecture = ImageArchitecture
```
Comment (dominikholler) [inline pkg/build/build.go:34]: Is there a way to infer the value of the destination container image architecture from the input artifacts architecture?

**Comment by dominikholler** [comment]
> /lgtm

> amd64 (https://github.com/opencontainers/image-spec/blob/main/config.md#properties uses GOARCH arch names) is the only supported arch in containerdisks at the moment so hard coding this in is fine as a bugfix. We obviously will want to change this if we ever support building arm64 images etc.

I did not recognize Lee's comment, I agree with him.

File: pkg/build/build.go (hunk 1 above)
Comment (rmohr) [inline pkg/build/build.go:34]: Woult it make sense to set the label also here with `cf.Labels` and only mutate the image config once?

Comment (0xFelix) [reply pkg/build/build.go:35]: Good call :+1:

**Comment by 0xFelix** [comment]
@lyarwood, @dominikholler Yes, for now only `amd64` containerdisks are built but we have to think about it in the future. (https://github.com/kubevirt/containerdisks/issues/41)

**Pull request merged**